liquibase-core/src/main/java/liquibase/Liquibase.java

```diff
@@ -316,6 +324,139 @@ public void update(Contexts contexts, LabelExpression labelExpression, boolean c
+            String mismatchMessage = String.format("   '%s' : mismatched DBMS value of %s", skippedChangeSet, dbmsList);
+            Scope.getCurrentScope().getLog(getClass()).info(mismatchMessage);
+
+            List<String> outputRow = new ArrayList<>();
+            outputRow.add(skippedChangeSet.toString());
+            outputRow.add(String.format("mismatched DBMS value of %s", dbmsList));
+            table.add(outputRow);
+        });
+
+        //
+        // Filtered because of labels or context
+        //
+        for (ChangeSetStatus st : filterDenied) {
+            st.getFilterResults().forEach(consumer -> {
+                String skippedMessage = String.format("   '%s' : %s", st.getChangeSet().toString(), consumer.getMessage());
+                Scope.getCurrentScope().getLog(getClass()).info(skippedMessage);
+
+                List<String> outputRow = new ArrayList<>();
+                outputRow.add(st.getChangeSet().toString());
+                outputRow.add(consumer.getMessage());
+                table.add(outputRow);
+            });
+        }
+
+        List<Integer> widths = new ArrayList<>();
+        widths.add(60);
+        widths.add(40);
+
+        OutputStream outputStream = new ByteArrayOutputStream();
+        Writer writer = createOutputWriter(outputStream);
+        TableOutput.formatOutput(table, widths, true, writer);
+        String outputTableString = outputStream.toString();
+        Scope.getCurrentScope().getUI().sendMessage(outputTableString);
+    }
+
+    private void showSummary(DatabaseChangeLog changeLog, StatusVisitor statusVisitor, List<ChangeSet> skippedChangeSets, List<ChangeSetStatus> filterDenied) {
+        Scope.getCurrentScope().getUI().sendMessage("");
+        int totalInChangelog = changeLog.getChangeSets().size() + skippedChangeSets.size();
+        int skipped = skippedChangeSets.size();
+        int filtered = filterDenied.size();
+        int totalAccepted = statusVisitor.getChangeSetsToRun().size();
+        int totalPreviouslyRun = totalInChangelog - filtered - skipped - totalAccepted;
+
+        String message = "UPDATE SUMMARY";
+        Scope.getCurrentScope().getLog(getClass()).info(message);
+        Scope.getCurrentScope().getUI().sendMessage(message);
+
+        message = String.format("Run:                     %6d", totalAccepted);
+        Scope.getCurrentScope().getLog(getClass()).info(message);
+        Scope.getCurrentScope().getUI().sendMessage(message);
+
+        message = String.format("Previously run:          %6d", totalPreviouslyRun);
+        Scope.getCurrentScope().getLog(getClass()).info(message);
+        Scope.getCurrentScope().getUI().sendMessage(message);
+
+        message = String.format("DBMS mismatch:           %6d", skipped);
+        Scope.getCurrentScope().getLog(getClass()).info(message);
+        Scope.getCurrentScope().getUI().sendMessage(message);
+
+        message = String.format("Not in filter:           %6d", filtered);
```

Comment:
Should we separate context vs label filtering into separate prompts?

Reply:
Maybe have more consistent wording between the reasons why it was skipped: Like:
```
UPDATE SUMMARY
Run:                          2
Previously run:               0
Skipped due to dbms:                1
Skipped due to labels:                0
Skipped due to contexts:                0
-------------------------------
Total change sets:            3
```
or
```
UPDATE SUMMARY
Run:                          2
Previously run:               0
Skipped due to:
    dbms:                1
    labels:                0
    contexts:                0
-------------------------------
Total change sets:            3
```
?

Reply:
This sounds reasonable.

Reply:
We have created another ticket for that